src/ui/public/share/directives/share_object_url.js

```diff
@@ -67,7 +67,11 @@ app.directive('shareObjectUrl', function (Private, Notifier) {
       };
 
       $scope.getUrl = function () {
-        return $location.absUrl();
+        let url = $location.absUrl();
+        if ($scope.shareAsEmbed) {
+          url = url.replace('?', '?embed=true&');
```

Comment:
This really isn't how we should be handling this. This fails when there isn't a query string and when there is a query string before the hash. This should instead use `require('url').parse` and `require('url').format`.